typescript/cli/src/rebalancer/config/Config.ts

```diff
@@ -0,0 +1,182 @@
+    for (const chainName of chainNames) {
+      const chain = config[chainName] as ChainConfig;
+
+      if (chain.override) {
+        for (const overrideChainName of Object.keys(chain.override)) {
+          // Each override key must reference a valid chain
+          if (!chainNames.has(overrideChainName)) {
+            ctx.addIssue({
+              code: z.ZodIssueCode.custom,
+              message: `Chain '${chainName}' has an override for '${overrideChainName}', but '${overrideChainName}' is not defined in the config`,
+              path: [chainName, 'override', overrideChainName],
+            });
+          }
+
+          // Override shouldn't be self-referencing
+          if (chainName === overrideChainName) {
+            ctx.addIssue({
+              code: z.ZodIssueCode.custom,
+              message: `Chain '${chainName}' has an override for '${chainName}', but '${chainName}' is self-referencing`,
+              path: [chainName, 'override', overrideChainName],
+            });
+          }
+        }
+      }
+    }
+  },
+);
+
+// Define separate types for each strategy config
+export type WeightedChainConfig = z.infer<typeof WeightedChainConfigSchema>;
+
+export type MinAmountChainConfig = z.infer<typeof MinAmountConfigSchema>;
+
+// Union type for all chain configs
+export type ChainConfig = z.infer<typeof ChainConfigSchema>;
+
+export type BaseConfig = z.infer<typeof BaseConfigSchema>;
+
+export class Config {
+  static load(
+    configFilePath: string,
+    rebalancerKey: string,
+    overrides: BaseConfig,
+  ) {
+    const config = readYamlOrJson(configFilePath);
+    const validationResult = ConfigSchema.safeParse(config);
+
+    if (!validationResult.success) {
+      throw new Error(fromZodError(validationResult.error).message);
+    }
+
+    const {
+      warpRouteId: fileWarpRouteId,
+      checkFrequency: fileCheckFrequency,
+      monitorOnly: fileMonitorOnly,
+      withMetrics: fileWithMetrics,
+      coingeckoApiKey: fileWithCoingeckoApiKey,
+      rebalanceStrategy: fileRebalanceStrategy,
+      ...chains
+    } = validationResult.data;
+
+    if (!Object.keys(chains).length) {
+      throw new Error('No chains configured');
+    }
+
+    const warpRouteId = overrides.warpRouteId ?? fileWarpRouteId;
+    const checkFrequency = overrides.checkFrequency ?? fileCheckFrequency;
+    const monitorOnly = overrides.monitorOnly ?? fileMonitorOnly ?? false;
+    const withMetrics = overrides.withMetrics ?? fileWithMetrics ?? false;
+    const coingeckoApiKey =
+      overrides.coingeckoApiKey ?? fileWithCoingeckoApiKey ?? '';
+    const rebalanceStrategy =
+      overrides.rebalanceStrategy ?? fileRebalanceStrategy;
+
+    if (!warpRouteId) {
+      throw new Error('warpRouteId is required');
+    }
+
+    if (!checkFrequency) {
+      throw new Error('checkFrequency is required');
+    }
+
+    if (!rebalanceStrategy) {
+      throw new Error('rebalanceStrategy is required');
+    }
```

Comment:
Why do we have these fields as optional if we then need them here? can't we update the schema to make them required?

Reply:
ooh, I think @ltyu already responded here: https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/6044#discussion_r2102922959

Reply:
yup I agree with him